### ISSUE TITLE
Duplicate variable names

### ISSUE THREAD

**Issue by arafath-mk** [kyleconroy/sqlc#1968]
### Version

1.12.0

### What happened?

In the generated function, there is a Go error. `id redeclared in this block`  

There are two id variables. One in the function parameter and another in the function body.

### Relevant log output

```shell
const getUserRole = `-- name: GetUserRole :one
SELECT id FROM public.user_roles
WHERE id = $1 LIMIT 1
`

func (q *Queries) GetUserRole(ctx context.Context, id int64) (int64, error) {
	row := q.db.QueryRow(ctx, getUserRole, id)
	var id int64
	err := row.Scan(&id)
	return id, err
}
```

### Database schema

```sql
CREATE TABLE user_roles (
  id   BIGSERIAL PRIMARY KEY
);
```

### SQL queries

```sql
-- name: GetUserRole :one
SELECT * FROM public.user_roles
WHERE id = $1 LIMIT 1;
```

### Configuration

```yaml
version: "2"
sql:
- schema: "server/database/postgres/migrate/"
  queries: "server/database/postgres/sqlc_queries/user/"
  engine: "postgresql"
  gen:
    go: 
      package: "user"
      out: "server/go/layer/repository/db/pgsql/user"
      sql_package: "pgx/v4"
```

### What operating system are you using?

Windows

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by zaneli** [comment]
Seems to fixed by #1298

**Comment by kyleconroy** [comment]
Duplicate of https://github.com/kyleconroy/sqlc/issues/1490
